`/scratch1/NCEPDEV/da/role.jedipara/CI/GDASApp/workflow/PR/1727/global-workflow/sorc/logs/build_gdas.log` contains
```
In file included from /scratch1/NCEPDEV/da/role.jedipara/CI/GDASApp/workflow/PR/1727/global-workflow/sorc/gdas.cd/bundle/gdas-utils/soca/diagb/gdas_soca_diagb.cc(1):
/scratch1/NCEPDEV/da/role.jedipara/CI/GDASApp/workflow/PR/1727/global-workflow/sorc/gdas.cd/bundle/gdas-utils/soca/diagb/gdas_soca_diagb.h(36): catastrophic error: cannot open source file "../gdas_soca_utils.h"
  #include "../gdas_soca_utils.h"
                                 ^

compilation aborted for /scratch1/NCEPDEV/da/role.jedipara/CI/GDASApp/workflow/PR/1727/global-workflow/sorc/gdas.cd/bundle/gdas-utils/soca/diagb/gdas_soca_diagb.cc (code 4)
make[2]: *** [gdas-utils/soca/diagb/CMakeFiles/gdas_soca_diagb.x.dir/build.make:76: gdas-utils/soca/diagb/CMakeFiles/gdas_soca_diagb.x.dir/gdas_soca_diagb.cc.o] Error 4
```

Directory `/scratch1/NCEPDEV/da/role.jedipara/CI/GDASApp/workflow/PR/1727/global-workflow/sorc/gdas.cd/bundle/gdas-utils/soca/diagb` contains
```
  -rw-r--r-- 1 role.jedipara da   248 May 30 16:41 CMakeLists.txt
  -rw-r--r-- 1 role.jedipara da  6497 May 30 16:41 README.md
  -rw-r--r-- 1 role.jedipara da   181 May 30 16:41 gdas_soca_diagb.cc
  -rw-r--r-- 1 role.jedipara da 10835 May 30 16:41 gdas_soca_diagb.h
  -rw-r--r-- 1 role.jedipara da  9093 May 30 16:41 gdas_soca_diagb_utils.h
```

The directory one level up contains
```
  -rw-r--r-- 1 role.jedipara da  1323 May 30 16:41 CMakeLists.txt
  drwxr-sr-x 2 role.jedipara da  4096 May 30 16:41 diagb
  -rw-r--r-- 1 role.jedipara da   197 May 30 16:31 gdas_ens_handler.cc
  -rw-r--r-- 1 role.jedipara da 17820 May 30 16:31 gdas_ens_handler.h
  -rw-r--r-- 1 role.jedipara da   201 May 30 16:31 gdas_incr_handler.cc
  -rw-r--r-- 1 role.jedipara da  3533 May 30 16:41 gdas_incr_handler.h
  -rw-r--r-- 1 role.jedipara da  7057 May 30 16:41 gdas_incr_qc.h
  -rw-r--r-- 1 role.jedipara da 18148 May 30 16:41 gdas_postprocincr.h
  -rw-r--r-- 1 role.jedipara da   220 May 30 16:31 gdas_socahybridweights.cc
  -rw-r--r-- 1 role.jedipara da  5677 May 30 16:31 gdas_socahybridweights.h
  -rw-r--r-- 1 role.jedipara da   253 May 30 16:31 gdassoca_obsstats.cc
  -rw-r--r-- 1 role.jedipara da  9278 May 30 16:31 gdassoca_obsstats.h
```

Neither contains `gdas_soca_utils.h`

--- a/utils/soca/gdas_soca_utils.h
+++ b/utils/soca/gdas_soca_utils.h
@@ -1,0 +1,74 @@
+#pragma once
+
+#include <cassert>
+#include <cmath>
+#include <vector>
+
+namespace gdasapp {
+namespace utils {
+
+  // Constants
+  constexpr double alpha = 2.0e-4;       // Thermal expansion coefficient [1/K]
+  constexpr double beta  = 7.6e-4;       // Haline contraction coefficient [1/psu]
+
+  /**
+   * @brief Computes ocean depth and bathymetry from ocean thickness values
+   *
+   * This function calculates:
+   * 1. Depth at of the center of each layers
+   * 2. Bathymetry as the total sum of ocean thickness values at each node
+   *
+   * @param viewHocn 2D array view containing ocean thickness values [nodes × levels]
+   *
+   * @return std::pair of ArrayViews containing:
+   *         - first: Computed depth values at each level [nodes × levels]
+   *         - second: Computed bathymetry values [nodes × 1]
+   */
+
+void computeDepthAndBathymetry(
+    const atlas::array::ArrayView<double, 2>& viewHocn,
+    atlas::array::ArrayView<double, 2>& viewDepth,
+    atlas::array::ArrayView<double, 2>& viewBathy) {
+
+    // Compute depth values
+    for (atlas::idx_t jnode = 0; jnode < viewDepth.shape(0); ++jnode) {
+        viewDepth(jnode, 0) = 0.5 * viewHocn(jnode, 0);
+        for (atlas::idx_t level = 1; level < viewDepth.shape(1); ++level) {
+            viewDepth(jnode, level) = viewDepth(jnode, level - 1)
+                                    + 0.5 * (viewHocn(jnode, level - 1) + viewHocn(jnode, level));
+        }
+    }
+
+    // Compute bathymetry values
+    for (atlas::idx_t jnode = 0; jnode < viewHocn.shape(0); ++jnode) {
+        viewBathy(jnode, 0) = std::accumulate(&viewHocn(jnode, 0),
+                                          &viewHocn(jnode, 0) + viewHocn.shape(1), 0.0);
+    }
+}
+
+/**
+ * @brief Compute steric height increment from temperature/salinity increments and layer thickness.
+ *
+ * @param tempIncr        Temperature increment profile [°C]
+ * @param saltIncr        Salinity increment profile [psu]
+ * @param layerThickness  Layer thickness profile [m]
+ * @return approximate steric height increment [m]
+ */
+inline double computeStericHeight(const std::vector<double> &tempIncr,
+                                  const std::vector<double> &saltIncr,
+                                  const std::vector<double> &layerThickness) {
+  assert(tempIncr.size() == saltIncr.size());
+  assert(saltIncr.size() == layerThickness.size());
+
+  double stericHeightIncr = 0.0;
+
+  for (size_t k = 0; k < tempIncr.size(); ++k) {
+    // dH = (-alpha * dT + beta * dS) * dz
+    stericHeightIncr += (alpha * tempIncr[k] - beta * saltIncr[k]) * layerThickness[k];
+  }
+
+  return stericHeightIncr;
+}
+
+}  // namespace utils
+}  // namespace gdasapp

